src/ddmd/ctfeexpr.d

```diff
@@ -990,18 +990,32 @@ extern (C++) int comparePointers(Loc loc, TOK op, Type type, Expression agg1, di
 }
 
 // True if conversion from type 'from' to 'to' involves a reinterpret_cast
-// floating point -> integer or integer -> floating point
-extern (C++) bool isFloatIntPaint(Type to, Type from)
+// between any combination of floating point and integral types, including
+// small static arrays (size permitting).
+extern (C++) bool isFloatIntPaint(Type from, Type to)
 {
-    return from.size() == to.size() && (from.isintegral() && to.isfloating() || from.isfloating() && to.isintegral());
+    const toUsed = (to.ty == Tfloat80) ? 80/8 : to.size();
```

Comment:
I know it's an unfortunate name, but `Tfloat80` is treated as largest supported floating point type in gdc and ldc.  `Tfloat80` can even be a double!